### DISCUSSION TITLE
Is it possible to disable "auto cd"?

### DISCUSSION THREAD

**Discussion by olalonde** [nvim-tree/nvim-tree.lua#2030] [Q&A]
I noticed that when I edit a file in a npm package, vim tree automatically changes directory to the root directory of the project (where the `package.json` file is located). Is it possible to disable this behavior? This is a bit annoying because I'm working on a mono repo with multiple npm packages and I would rather have vim tree auto cd to the root .git directory or not auto cd at all. Thanks.

**Comment by alex-courtis** [comment]
This is indeed undesirable behaviour, however nvim-tree will not do this by default.

What configuration have you set? `update_focused_file` and `sync_root_with_cwd` and `actions.change_dir` affect such behaviour.

**Reply by olalonde** [reply]
Thanks, I just realized it was setup in my config:

```lua
nvim_tree.setup {
  update_focused_file = {
    enable = true,
    update_cwd = true,
  },
}
```

It was sometimes useful though, is it possible to trigger it manually? e.g. `:NvimTreeUpdateCwd` or something like that? Thanks.

**Reply by alex-courtis** [reply]
My workflow is simply to `:cd new/path` with `sync_root_with_cwd` enabled. That will trigger the tree root change to vim's new cwd.

**Reply by olalonde** [reply]
Thanks but I don't want that happening automatically because I often work in a monorepo while editing multiple packages at the same time and I don't want Nerd Tree to change every time I edit a file. It would be nice to be able to trigger that behavior manually though because it is sometimes useful. I'm not sure if it would require to implement a new command on this plugin (e.g. something like `:NVimTreeSyncRootWithCwd`).

**Reply by alex-courtis** [reply]
I see. That already exists:

`:help nvim-tree.api.tree.change_root()`

There are a couple of related API functions that you might find interesting.

**Reply by olalonde** [reply]
Sorry not too familiar with Vim scripting. I tried to type `:nvim-tree.api.tree.change_root()` and it obviously didn't work. I guess I would have to expose that function as an ex command in my config files? Thanks!

**Reply by alex-courtis** [reply]
As an ex command:
`:lua require("nvim-tree.api").tree.change_root("/home/me")`

See `:help nvim-tree-mappings` and [wiki: on_attach](https://github.com/nvim-tree/nvim-tree.lua/wiki/Migrating-To-on_attach)
